backend/pkg/api/handlers/common.go

```diff
@@ -724,108 +722,121 @@ func isUserAdmin(user *types.UserInfo) bool {
 		Error: err.Error(),
 	}
-	writeResponse(w, code, response)
+	writeResponse(w, r, code, response)
 }
 
-func returnOk(w http.ResponseWriter, data interface{}) {
-	writeResponse(w, http.StatusOK, data)
+func returnOk(w http.ResponseWriter, r *http.Request, data interface{}) {
+	writeResponse(w, r, http.StatusOK, data)
 }
 
-func returnCreated(w http.ResponseWriter, data interface{}) {
-	writeResponse(w, http.StatusCreated, data)
+func returnCreated(w http.ResponseWriter, r *http.Request, data interface{}) {
+	writeResponse(w, r, http.StatusCreated, data)
 }
 
-func returnNoContent(w http.ResponseWriter) {
-	writeResponse(w, http.StatusNoContent, nil)
+func returnNoContent(w http.ResponseWriter, r *http.Request) {
+	writeResponse(w, r, http.StatusNoContent, nil)
 }
 
 // Errors
 
-func returnBadRequest(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusBadRequest, err)
+func returnBadRequest(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusBadRequest, err)
 }
 
-func returnUnauthorized(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusUnauthorized, err)
+func returnUnauthorized(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusUnauthorized, err)
 }
 
-func returnNotFound(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusNotFound, err)
+func returnNotFound(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusNotFound, err)
 }
 
-func returnConflict(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusConflict, err)
+func returnConflict(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusConflict, err)
 }
 
-func returnForbidden(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusForbidden, err)
+func returnForbidden(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusForbidden, err)
 }
 
-func returnTooManyRequests(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusTooManyRequests, err)
+func returnTooManyRequests(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusTooManyRequests, err)
 }
 
-func returnGone(w http.ResponseWriter, err error) {
-	returnError(w, http.StatusGone, err)
+func returnGone(w http.ResponseWriter, r *http.Request, err error) {
+	returnError(w, r, http.StatusGone, err)
 }
 
-func returnInternalServerError(w http.ResponseWriter, err error) {
-	log.Error(err, "internal server error", 2, nil)
+const maxBodySize = 10 * 1024
+
+func logApiError(r *http.Request, err error) {
+	body, _ := io.ReadAll(io.LimitReader(r.Body, maxBodySize))
+	log.Error(err, "error handling request", 3, nil,
+		map[string]interface{}{
+			"endpoint": r.Method + " " + r.URL.Path,
+			"query":    r.URL.RawQuery,
+			"body":     string(body),
+		},
+	)
+}
+
+func returnInternalServerError(w http.ResponseWriter, r *http.Request, err error) {
+	logApiError(r, err)
```

Comment:
This is prob the most significant change

Reply:
The fixed `callerSkip` value of 3 might lead to unclear logging depending on how the error is handled. If handled via `handleErr` a depth of 4 would be required to log the expected function in the call stack.

Could solve this by forbidding all `returnXXX` shortcuts in all handlers (and always using `handleErr(w, r, newXXXErr("<error>"))`) or by passing the `callerSkip` to `logApiError`

Reply:
You're right. Added the `callerSkip` param. `returnInternalServerError` should be used either only by `handleErr` or only by handlers directly. Since I don't want to prevent handlers from using these quick return funcs, I'll choose the latter. And because no handler uses `returnInternalServerError`, I removed it.